orchestrator/src/compression/blob.rs

```diff
@@ -0,0 +1,303 @@
+            da_word |= BigUint::from(new_nonce) << 188;
+        }
+
+        // Determine if we need 8 or 64 bits for num_changes
+        let needs_large_updates = num_changes >= 256;
+
+        if needs_large_updates {
+            // Use 64 bits for updates
+            da_word |= BigUint::from(num_changes) << 124;
+        } else {
+            // Use 8 bits for updates
+            da_word |= BigUint::from(num_changes) << 180;
+            da_word |= BigUint::one() << 123;
+        }
+
+        // Class flag
+        da_word |= if class_flag { BigUint::one() << 122 } else { BigUint::zero() };
+    } else {
+        // Old format (pre-0.13.3)
+        //
+        // Binary encoding layout (252 bits total):
+        //
+        // Bit positions:
+        // 251                                                                                   0
+        // |                                                                                     |
+        // v                                                                                     v
+        // ┌───────────────────────────────────────────────────────┬─────┬─────────┬─────────────┐
+        // │                 Zero Padding                          │cls  │  nonce  │ num_changes │
+        // │                  (123 bits)                           │flg  │(64 bits)│ (64 bits)   │
+        // │                  [251:129]                            │[128]│[127:64] │  [63:0]     │
+        // └───────────────────────────────────────────────────────┴─────┴─────────┴─────────────┘
+        //
+        // Field breakdown:
+        // - Zero Padding (123 bits)  : [251:129] - Padded with zeros
+        // - class_flag (1 bit)       : [128]     - Class type flag
+        // - nonce (64 bits)          : [127:64]  - Nonce value
+        // - num_changes (64 bits)    : [63:0]    - Number of changes
+        //
+        // Total used bits: 123 + 1 + 64 + 64 = 252 bits (exactly)
+        //
+
+        // Class flag
+        da_word |= if class_flag { BigUint::one() << 128 } else { BigUint::zero() };
+
+        // Nonce
+        if let Some(new_nonce) = nonce_change {
+            let new_nonce = new_nonce.to_u64().expect("nonce too big");
+            da_word |= BigUint::from(new_nonce) << 64;
+        }
+
+        // Number of changes
+        da_word |= BigUint::from(num_changes);
+    }
+
+    Ok(Felt::from(da_word))
+}
+
+/// Converts a vector of felt into a vector of bigUint
+/// The output length depends on the input length (ceil(input_len / BLOB_LEN) * BLOB_LEN)
+pub fn convert_to_biguint(elements: &[Felt]) -> Vec<BigUint> {
+    let num_blocks = (elements.len() + BLOB_LEN - 1) / BLOB_LEN;
+    let output_len = num_blocks * BLOB_LEN;
+    let pad_len = output_len - elements.len();
+
+    elements
+        .iter()
+        .map(|e| BigUint::from_bytes_be(&e.to_bytes_be()))
+        .chain(std::iter::repeat(BigUint::zero()).take(pad_len))
+        .collect()
+}
```

Comment:
let's divide this function in two parts for separation of concern, we can test these individually as well then!
```suggestion
/// Converts felt elements to BigUint and pads to block boundaries
/// 
/// Each block has BLOB_LEN elements. If the input doesn't fill complete blob,
/// the output is padded with zeros to the next blob boundary.
pub fn convert_to_biguint(elements: &[Felt]) -> Vec<BigUint> {
    let converted: Vec<BigUint> = elements
        .iter()
        .map(|e| BigUint::from_bytes_be(&e.to_bytes_be()))
        .collect();
    
    pad_to_block_size(converted, BLOB_LEN)
}

fn pad_to_blob_size(input: Vec<BigUint>, blob_size: usize) -> Vec<BigUint> {
    let num_blocks = (input.len() + blob_size - 1) / blob_size;
    let target_len = num_blobs * blob_size;
    let pad_len = target_len - input.len();

    input.into_iter()
        .chain(std::iter::repeat(BigUint::zero()).take(pad_len))
        .collect()
}
```